src/ConventionalCommitUtilities.js

```diff
@@ -1,108 +1,186 @@
 "use strict";
 
+const conventionalChangelogCore = require("conventional-changelog-core");
+const conventionalRecommendedBump = require("conventional-recommended-bump");
 const dedent = require("dedent");
+const getStream = require("get-stream");
 const log = require("npmlog");
+const npa = require("npm-package-arg");
 const path = require("path");
 const semver = require("semver");
 
-const ChildProcessUtilities = require("./ChildProcessUtilities");
 const FileSystemUtilities = require("./FileSystemUtilities");
+const ValidationError = require("./utils/ValidationError");
 
-const CHANGELOG_NAME = "CHANGELOG.md";
-const CHANGELOG_HEADER = dedent(`# Change Log
+const cfgCache = new Map();
 
-  All notable changes to this project will be documented in this file.
-  See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.`);
+function getChangelogConfig(changelogPreset = "conventional-changelog-angular") {
```

Comment:
We could just let [`conventional-changelog-preset-loader`](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-preset-loader#usage) do the dirty work, but then we wouldn't get caching and more flexibility with scopes and such.